drivers/rilmodem/voicecall.c

```diff
@@ -372,19 +377,29 @@ static void rild_cb(struct ril_msg *message, gpointer user_data)
 
 	g_ril_print_response_no_args(vd->ril, message);
 
-	/* CLCC will update the oFono call list with proper ids  */
-	if (!vd->clcc_source)
+	/*
+	 * CLCC will update the oFono call list with proper ids. However, if
+	 * the callback isn't there, it's likely that an
+	 * UNSOL_RESPONSE_CALL_STATE_CHANGED has been issued and the CLCC
+	 * has been called already. So, there's no need to trigger another CLCC.
+	 */
+	if (!vd->clcc_source && vd->cb)
 		vd->clcc_source = g_timeout_add(POLL_CLCC_INTERVAL,
 						ril_poll_clcc, vc);
-
-	/* we cannot answer just yet since we don't know the call id */
-	vd->cb = cb;
-	vd->data = cbd->data;
-
 	return;
 
 out:
-	cb(&error, cbd->data);
+	/*
+	 * If the callback isn't there, it's likely that the callback has been
+	 * called with success already in CLCC's response, so, we can't signal
+	 * the error to the caller. However, it's unlikely to have the new call
+	 * in the call list but the dial itself fails.
+	 */
+	if (vd->cb) {
+		cb(&error, cbd->data);
```

Comment:
This situation is weird enough to deserve an ofono_error call so we can spot it in syslog.